src/transformers/models/bloom/modeling_bloom.py

```diff
@@ -0,0 +1,990 @@
+# coding=utf-8
+# Copyright 2022 HuggingFace Inc. team and BigScience workshop.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""PyTorch BLOOM model."""
+
+import math
+from typing import Tuple
+
+import torch
+import torch.nn.functional as F
+import torch.utils.checkpoint
+from torch import Tensor, nn
+from torch.nn import CrossEntropyLoss, LayerNorm
+
+from ...file_utils import add_code_sample_docstrings, add_start_docstrings, add_start_docstrings_to_model_forward
+from ...modeling_outputs import BaseModelOutputWithPastAndCrossAttentions, CausalLMOutputWithCrossAttentions
+from ...modeling_utils import Conv1D, PreTrainedModel
+from ...utils import logging
+from .configuration_bloom import BloomConfig
+from .fused_bias_gelu import bias_gelu_impl
+
+
+logger = logging.get_logger(__name__)
+
+_CHECKPOINT_FOR_DOC = "bigscience/Bloom"
+_CONFIG_FOR_DOC = "BloomConfig"
+_TOKENIZER_FOR_DOC = "BloomTokenizer"
+
+BLOOM_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "bigscience/bigscience-small-testing",
+    "bigscience/bloom-350m",
+    "bigscience/bloom-760m",
+    "bigscience/bloom-1b3",
+    "bigscience/bloom-2b5",
+    "bigscience/bloom-6b3",
+    "bigscience/bloom-176b",
+]
+
+
+# Utility functions below:
+def pre_process_alibi_for_pad(alibi, attention_mask, num_heads):
```

Comment:
Is this not used? I didn't find any usage for this.

Reply:
Yes my initial intention was to shift the alibi embeddings when passing padded inputs. But it appeared that it didn't affected that much the generations.

Reply:
Ah right, actually this makes sense. otherwise your model doesn't behave the same when a sample is in a different batch.

Reply:
But actually the problem I had with that is that a lot of tests with random attention masks were not passing, that is why I didn't moved further in this direction

Reply:
Yeah but then I think you should fix them as you might discover new bugs. I think this is needed to have logits match in the batched case.